### PR TITLE
Pull Request for Issue 2377: Add additional functionality to PGM scan configuration view.

### PR DESCRIPTION
**Pull request by mButlerDM**
Added in a detector selection component similar to the one used by BioXAS.

Setup the export options and added a check box to enable export, default is enabled.

Adjusted visual components to be cleaner. 

Corrected double entry in pri file.

Corrected component load order in PGMBeamline to load setupControlSets later in the list to resolve a segfault on launch.

#2377

### CODE REVIEW

**Comment by mButlerDM** [comment]
@helfrij Could you review when you have a chance? Thanks!

File: source/application/PGM/PGMAppController.cpp [hunk 1]
```diff
@@ -137,7 +141,9 @@ void PGMAppController::registerDBClasses()
 
 void PGMAppController::registerExporterOptions()
 {
-
+        AMExporterOptionXDIFormat *pgmXASExportOptions = PGM::buildXDIFormatExporterOption("PGMXASDefault", true);
```
Comment (helfrij) [inline source/application/PGM/PGMAppController.cpp]: In some of the scan configurations on other beamlines, we make an additional call to build exporter options in the scan configuration constructor, pretty much just repeating these calls made here. Just wanted to check whether that was needed on PGM as well.

**Comment by helfrij** [comment]
Looks good, @mButlerDM. One minor question.

Time Estimate: 10 minutes.

**Comment by dretrex** [comment]
Generally, this looks good.  @helfrij brings up a good point, which should be addressed.  Also, there are some indent issues.

Time Estimate: 1 minute

**Pull request merged**